src/main/kotlin/extentions/NotificationsExtentions.kt

```diff
@@ -1,13 +1,32 @@
 package co.novu.extensions
 
 import co.novu.Novu
+import co.novu.dto.request.NotificationRequest
 import kotlinx.coroutines.runBlocking
 import mu.KotlinLogging
+import java.math.BigInteger
 
 private val logger = KotlinLogging.logger {}
 
+@Deprecated("Use notifications(notificationRequest: NotificationRequest)")
 fun Novu.notifications(channels: List<String>? = null, templates: List<String>? = null, emails: List<String>? = null, search: String? = null, page: String? = null, transactionId: String? = null) = runBlocking {
-    val response = notificationsApi.getNotifications(channels, templates, emails, search, page, transactionId)
+    val response = notificationsApi.getNotifications(channels, templates, emails, search, page?.let { BigInteger(it) }, transactionId)
+    if (response.isSuccessful) {
+        response.body().apply { logger.info { this } }
+    } else {
+        throw Exception(response.errorBody()?.string())
+    }
+}
+
+fun Novu.notifications(notificationRequest: NotificationRequest) = runBlocking {
+    val response = notificationsApi.getNotifications(
+        channels = notificationRequest.channels,
+        templates = notificationRequest.templates,
+        emails = notificationRequest.emails,
+        search = notificationRequest.search,
+        page = notificationRequest.page,
+        transactionId = notificationRequest.transactionId
+    )
     if (response.isSuccessful) {
         response.body().apply { logger.info { this } }
```

Comment:
Just to catch this now, this log should be a verbose log and not an info log at these points.
You did a good job adding logging as most people would have not done this.

I am concerned if we continue doing this as info then we risk adding to much log outputs in applications just for Novu calls when it should be about there application.